src/Uno.UI/UI/Xaml/UIElement.cs

```diff
@@ -910,6 +910,45 @@ public void InvalidateArrange()
 		}
 #endif
 
+		/// <summary>
+		/// This method has to be invoked for element that are going to be recycled WITHOUT necessarily being unloaded / loaded.
+		/// For instance, this is is not expected to be invoked for elements recycled by the the template pool as they are always unloaded.
+		/// The main use case is for ListView and is expected to be invoked by the ListView.CleanUpContainer.
+		/// </summary>
+		/// <remarks>This will walk the tree down to invoke this on all children!</remarks>
+		internal static void PrepareForRecycle(object view)
+		{
+			if (view is UIElement elt)
+			{
+				elt.PrepareForRecycle();
+			}
+			else
+			{
+				foreach (var child in VisualTreeHelper.GetManagedVisualChildren(view))
+				{
+					child.PrepareForRecycle();
+				}
+			}
+		}
+
+		/// <summary>
+		/// This method has to be invoked on element that are going to be recycled WITHOUT necessarily being unloaded / loaded.
+		/// For instance, this is is not expected to be invoked for elements recycled by the the template pool as they are always unloaded.
+		/// The main use case is for ListView and is expected to be invoked by the ListView.CleanUpContainer.
+		/// </summary>
+		/// <remarks>This will walk the tree down to invoke this on all children!</remarks>
```

Comment:
```suggestion
		/// This method has to be invoked on elements that are going to be recycled WITHOUT necessarily being unloaded / loaded.
		/// For instance, this is not expected to be invoked for elements recycled by the template pool as they are always unloaded.
		/// The main use case is for ListView and is expected to be invoked by ListView.CleanUpContainer.
		/// </summary>
		/// <remarks>This will walk the tree down to invoke this on all children!</remarks>
```